relay-server/src/extractors/remote.rs

```diff
@@ -0,0 +1,73 @@
+//! Extractors for types from other crates via [`Xt`].
+
+use axum::extract::{FromRequest, Request};
+use axum::http::StatusCode;
+use axum::response::{IntoResponse, Response};
+use multer::Multipart;
+
+use crate::service::ServiceState;
+use crate::utils::{self, ApiErrorResponse};
+
+/// A transparent wrapper around a type that implements [`FromRequest`] or [`IntoResponse`].
+///
+/// # Example
+///
+/// ```ignore
+/// use std::convert::Infallible;
+///
+/// use axum::extract::{FromRequest, Request};
+/// use axum::response::IntoResponse;
+///
+/// use crate::extractors::Xt;
+///
+/// // Derive `FromRequest` for `bool` for illustration purposes:
+/// #[axum::async_trait]
+/// impl<S> axum::extract::FromRequest<S> for Xt<bool> {
+///     type Rejection = Xt<Infallible>;
+///
+///     async fn from_request(request: Request) -> Result<Self, Self::Rejection> {
+///         Ok(Xt(true))
+///     }
+/// }
+///
+/// impl IntoResponse for Xt<Infallible> {
+///    fn into_response(self) -> axum::response::Response {
+///        match self.0 {}
+///    }
+/// }
+/// ```
+#[derive(Debug)]
+pub struct Xt<T>(pub T);
```

Comment:
nit: Is `Xt` a common abbreviation for helper types like this? If not, should we call it `Wrapper`?

Reply:
It's definitely not a common name, I chose it as an abbreviation for "extractor" mostly because it's short. The intended use is to immediately destructure it. "Wrapper" to me just sounded less specific to what this type is meant for. 

Should we name it `Extract` or `Extractor`?

Reply:
`Extractor` sound good!

Reply:
Just realizing that we would need the same for rejections. So either we go with something more generic indeed, or we have two types for that...